src/code_translation/ir/nodes.py

```diff
@@ -108,15 +123,20 @@ def to_cst_translation(self, sql_access_method) -> CSTTranslation:
             # TODO: DataFrameNode needs key also aka 'index_col' in read_sql
             left_table_alias = "S1"
             right_table_alias = "S2"
-            query_str = f"SELECT * FROM ({left_sql}) AS {left_table_alias} JOIN ({right_sql}) AS {right_table_alias} ON {left_table_alias}.{left_key} = {right_table_alias}.{right_key}"
-
+            return f"SELECT * FROM ({left_sql}) AS {left_table_alias} {join_operator} ({right_sql}) AS {right_table_alias} ON {left_table_alias}.{left_key} = {right_table_alias}.{right_key}"
```

Comment:
left_key and right_key are only both defined if left_set_key and right_set_key are true, right? because the condition only checks if one of them is true

Reply:
you are right